### ISSUE TITLE
Update computron system & script

### ISSUE THREAD

**Issue by jcfr** [Slicer/DashboardScripts#52]
Since the macOS 10.15 (Catalina) is obsolete, the system should be updated.

* System updates
  * [x] Update system from macOS 10.15 (Catalina) to at least macOS 12.0 (Monterrey) :heavy_check_mark: 
  * [x] Install XCode :heavy_check_mark: 
  * [x] Rebuild Qt 5.15.2 using updated XCode
    * [ ] Update qt-easy-build to support building on macOS 13. See https://github.com/jcfr/qt-easy-build/pull/69

* Script updates [^1][^2]. See https://github.com/Slicer/DashboardScripts/pull/53
  * [x] Update `COMPILER` version
  * [x] Update `CMAKE_OSX_DEPLOYMENT_TARGET`
  * [x] Remove obsolete `factory-south-macos-*` scripts 

Related issues and pull-requests:
* https://github.com/Slicer/Slicer/pull/6695

[^1]: https://github.com/Slicer/DashboardScripts/blob/2280c47d6633b9cc155fa412cf6f647afc083717/computron-slicer_preview_nightly.cmake
[^2]: https://github.com/Slicer/DashboardScripts/blob/2280c47d6633b9cc155fa412cf6f647afc083717/computron-slicerextensions_preview_nightly.cmake

**Comment by jcfr** [comment]
The operating system has been updated to macOS 13 (Ventura)

```
$ sw_vers   
ProductName:         macOS
ProductVersion:      13.0.1
BuildVersion:        22A400
```

**Comment by jcfr** [comment]
| | |
|--|--|
|![image](https://user-images.githubusercontent.com/219043/211175015-57002ecb-3cf0-44a4-a0d6-1685b6b03457.png) | ![image](https://user-images.githubusercontent.com/219043/211175044-64c0cb67-4c5c-472c-be23-e5f1de7a09d4.png) |

Since current version still reports `11.7`:

```
$ xcodebuild -version  
Xcode 11.7
Build version 11E801a

$ xcode-select -p
/Applications/Xcode_11.7.app/Contents/Developer
```

... Xcode version 14.2 installation folder was first renamed and then set as default using `xcode-select`:

```
$ sudo mv /Applications/Xcode.app /Applications/Xcode-14.2.app

$ sudo xcode-select -s /Applications/Xcode-14.2.app/Contents/Developer

$ xcode-select -p                                            
/Applications/Xcode-14.2.app/Contents/Developer

$ xcodebuild -version 
Xcode 14.2
Build version 14C18

$ clang --version 
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin22.1.0
Thread model: posix
InstalledDir: /Applications/Xcode-14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by jcfr** [comment]
Command used to build Qt 5.15.2 is the following:

```
cd /D/Support/qt-easy-build
git checkout 5.15.2
cd /D/Support/
./qt-easy-build/Build-qt.sh -s macosx13.1 -d 11 -j 4
```

Additionally patch adapted from https://github.com/microsoft/vcpkg/issues/21055 was also integrated to workaround build issue.

Based on my understanding the target deployement platform specified when building Qt (macOS 11) is consistent with https://github.com/Slicer/Slicer/pull/6695

@jamesobutler Did I miss anything ?

**Comment by jamesobutler** [comment]
I think you are doing everything that is needed. Will ultimately need to issue the PR in this repo to update things like `factory-south-macos-slicer_preview_nightly.cmake` to `computron-slicer_preview_nightly.cmake` as part of your listed "Script Updates" to-do list.

If you can utilize the changes in https://github.com/Slicer/Slicer/pull/6695 to do a sanity test to make sure "11.0" is correct would be helpful. It's a little weird with the change of the yearly release being the major version number now.

**Comment by jcfr** [comment]
> re: patch Qt 5.15.2

Turns out the fixes has already been included in the [qt/qtbase@5.15](https://github.com/qt/qtbase/tree/5.15) branch.

See https://github.com/qt/qtbase/commit/8467beddb7239cc213ae13900fa30e3d26df5e78
